### PR TITLE
DataViewModel Supports Multiple, Settable SortBy and GroupBy Fields

### PR DESCRIPTION
**Pull request by petradish**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [N/A] Reviewed and tested on Mobile, or determined not required.
- [N/A] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

This PR addresses https://github.com/xh/hoist-react/issues/1681
DataViewModel now pushes all store fields as columns but renders only the `sortBy` field as visible. This allows for multiple sorters and groupings as well as making them settable through GridModel's trampolined setter methods.

### CODE REVIEW

File: cmp/dataview/DataViewModel.js [hunk 1]
```diff
@@ -88,26 +85,26 @@ export class DataViewModel {
...
+        // single visible column to the desired sort field while other fields are hidden columns to allow
+        // for multiple sorters and groupings or using setGroupBy and setSortBy methods on the model itself.
 
-        const columns = [{
-            field,
+        const columns = store.fields.filter(field => {
+            return field !== sortBy;
+        }).map(field => {
+            return {
+                colId: field,
```
Comment (petradish) [inline cmp/dataview/DataViewModel.js]: Not sure if I need colIds here, as field seems to be sufficient identification (at least in the toolbox examples)

**Comment by lbwexler** [comment]
Petra and I working through this one.

File: cmp/dataview/DataViewModel.js [hunk 2]
```diff
@@ -88,26 +85,22 @@ export class DataViewModel {
...
-        // single column to the desired sort field. (The field setting should not have any other
-        // meaningful impact on the grid, since we use a custom renderer and mark it as complex to
-        // ensure each item re-renders on any record change.)
-        let field = 'id';
-        if (sortBy.length === 1) {
-            let sorter = sortBy[0];
-            if (!(sorter instanceof GridSorter)) sorter = GridSorter.parse(sorter);
-            field = sorter.colId;
-        }
+        // single visible column to the desired sortBy field while other fields are hidden columns to allow
```
Comment (lbwexler) [inline cmp/dataview/DataViewModel.js]: comment out of date now

File: cmp/grid/Grid.js [hunk 3]
```diff
@@ -201,6 +201,7 @@ class LocalModel {
             },
             popupParent: document.querySelector('body'),
             suppressAggFuncInHeader: true,
+            suppressMakeColumnVisibleAfterUnGroup: true,
```
Comment (lbwexler) [inline cmp/grid/Grid.js]: I thought a bit more about this -- let's just apply this now in DataView's agOptions, where we know we absolutely want this, and that will allow us to move forward with our DataView enhancements.

Think changing Grid default itself needs to be done with a bit more care, testing, and consideration of what we want for our grouping API.

Comment (petradish) [reply cmp/grid/Grid.js]: Great, I moved it to DataView only and its localModel now accepts props and agOptions defined as PT to allow for overriding of the suppressMakeColumnVisibleAfterUngroup prop

**Comment by lbwexler** [comment]
Discussed with Petra - she will enhance toolbox with better examples here, but looking forward to getting this in to v30 to more fully round-out the API changes to DataView/

**Pull request merged**